addons/beehave/nodes/beehave_tree.gd

```diff
@@ -27,25 +27,31 @@ signal tree_disabled
 		return enabled
 
 @export_node_path var actor_node_path : NodePath
-@export var blackboard:Blackboard
+@export var blackboard:Blackboard:
+	set(b):
+		blackboard = b
+		if blackboard and internal_blackboard:
+			remove_child(internal_blackboard)
+			internal_blackboard.free()
+			internal_blackboard = null
+		elif not blackboard and not internal_blackboard:
+			internal_blackboard = Blackboard.new()
+			add_child(internal_blackboard)
 
 var actor : Node
 var status : int = -1
+var internal_blackboard: Blackboard
 
 var _process_time_metric_name : String
 var _process_time_metric_value : float = 0.0
 
 func _ready() -> void:
 	if Engine.is_editor_hint():
 		return
-	
-	if self.get_child_count() != 1:
-		push_warning("Beehave error: Root %s should have one child (NodePath: %s)" % [self.name, self.get_path()])
-		disable()
-		return
 		
 	if not blackboard:
-		blackboard = Blackboard.new()
+		internal_blackboard = Blackboard.new()
+		add_child(internal_blackboard)
```

Comment:
`add_child` is in context of the BeehaveTree, right? Will this cause any errors or warnings, since it only wants to have one child?

Reply:
Actually, add_child has an argument that adds it to the "inner scope" that I will utilise! 

https://docs.godotengine.org/en/latest/classes/class_node.html#class-node-method-add-child